pkg/datapath/linux/ipsec/ipsec_linux.go

```diff
@@ -545,12 +546,17 @@ func loadIPSecKeys(r io.Reader) (int, uint8, error) {
 		// we remove any stale keys for example when a restart changes keys.
 		// In the restart case oldSpi will be '0' and cause the delete logic
 		// to run.
-		if oldSpi != ipSecKey.Spi {
+		if oldKey == nil || oldKey.Spi != ipSecKey.Spi {
 			go func() {
 				time.Sleep(linux_defaults.IPsecKeyDeleteDelay)
 				scopedLog.Info("New encryption keys reclaiming SPI")
 				ipsecDeleteXfrmSpi(ipSecKey.Spi)
 			}()
+		} else if !compareIPsecKeys(oldKey, ipSecKey) {
+			// If the keys have changed this would indicate an attempted rotation without proper version
+			// management as the SPI has not changed.
+			return 0, 0, fmt.Errorf("The IPSec keys with SPI %d have been changed without incrementing the key id.\n"+
+				"For instructions on rotating keys, see https://docs.cilium.io/en/latest/gettingstarted/encryption/#key-rotation", oldKey.Spi)
```

Comment:
I ran some tests and this doesn't work. We never see this error.

`loadIPSecKeys` is only called at startup, so as far as I can see, `oldKey` will always be `nil`. Instead, we should probably retrieve the `oldKey` [from the kernel](https://pkg.go.dev/github.com/vishvananda/netlink#XfrmStateList).